scripts/run_frontend_tests.sh

```diff
@@ -55,7 +55,7 @@ maybeInstallDependencies "$@"
 # and merges them all to a single file which Karma uses to run its tests. The
 # Karma is unable to run the tests on multiple files and the DI fails in that
 # case, the reason of which is unclear.
-find . -type f \( -name "*.spec.ts" -o -name "*Spec.ts" \) -exec cat {} \; > ./core/templates/dev/head/combined-tests.spec.ts
+find . -type f \( -name "*.spec.ts" ! -name "solution-verification.service.spec.ts" ! -name "state-name-editor.directive.spec.ts" ! -name "state-content-editor.directive.spec.ts" ! -name "state-interaction-editor.directive.spec.ts" ! -name "combined-tests.spec.ts" -o -name "*Spec.ts" ! -name "MusicNotesInputSpec.ts" \) -exec cat {} \; > ./core/templates/dev/head/combined-tests.spec.ts
```

Comment:
Why is the "-o" part in the middle of the command?

I think you need comments to explain the exclusion here, too. I am also concerned about having this list in multiple places, because the two lists are going to skew. Can you extract it from the Karma config file instead?

Reply:
I have removed the list completely from the karma conf since the files would not be collected by karma but instead would be done by this.

The logic of having the `-o` part in the middle is that the command works that way; i.e. you have a pattern to match against and the patterns to exclude in one place. So *.spec.ts matched will exclude the succeeding filename suffixed by `! -name`. The next list to be generated is done by suffixing `-o` to it. *Spec.ts will not collect *.spec.ts files so those files will never be included.

Re the explanation, I have explained the same in the above comments. Did you take a look at them? Please suggest any improvements if any.

Reply:
Ah, ok, thanks for explaining. Shouldn't we be getting rid of all the Spec.ts files though? I thought we should be following a spec.ts naming convention everywhere.

In terms of the explanation, I think we need more explanation about the "known failing files". Add links to issue threads about (a) the known failing files, and (b) Karma being unable to run tests on multiple files. In those issue threads please describe the issue really clearly in detail so that it is clear what files to look at, what has been tried already, and how the fixer will know that the issue is fixed.

Finally, I just realized that this might make the frontend coverage checks even more problematic ... /cc @vojtechjelinek @apb7 @ankita240796 for info.

Reply:
Re the naming, I have created an issue for this in the past to have the remaining directories under the new convention -> https://github.com/oppia/oppia/issues/6975

I have created the issue for the multiple spec files problem -> https://github.com/oppia/oppia/issues/7053 

Updated the comment too.

Re frontend coverage check, I don't have much idea regarding this. @apb7 might provide some insight.

Reply:
Thanks!

Hm, I am a bit confused about why #6975 still exists. I thought you had finished that completely as part of the prerequisite work, or were planning too at least?

If not, what is the plan?

Reply:
I had finished the `pages/`, `components/` and `filters/` part since they were part of the folders by feature PR. The other directories were alredy sorted in folders-by-feature so I did not cover them as the naming convention PRs went in along with the folders by features PR.

I plan to change these names as I upgrade them. Or at the time of modularity. I'll decide and let you know.